tools/batch_translate.py

```
@@ -0,0 +1,221 @@
+    # Add any additional translator arguments
+    if translator_args:
+        cmd.extend(translator_args)
+    
+    logger.info(f"Translating: {file_path.name} -> {output_file.name}")
+    
+    if dry_run:
+        logger.info(f"DRY RUN: Would execute: {' '.join(cmd)}")
+        return True
+    
+    try:
+        result = subprocess.run(cmd, capture_output=True, text=True, check=True)
```

Ruff (0.11.9)

`subprocess` call: check for execution of untrusted input

(S603)